### PR TITLE
Format Admin Dashboard Closes #47

### PR DESCRIPTION
**Pull request by scottalexandra**

### CODE REVIEW

**Comment by scottalexandra** [comment]
formatted dashboard according to the wireframe
Added Cms instructions to the admin dashboard
add company count by status
added companies to seed file

File: app/models/company.rb [hunk 1]
```diff
@@ -13,4 +13,8 @@ class Company < ActiveRecord::Base
   scope :dead, -> { where status: "dead"}
 
   STATUS_OPTIONS = { "Uncontacted" => "uncontacted", "Contacted" => "contacted", "Confirmed" => "confirmed", "Dead" => "dead" }
+
+  def self.company_count_by_status(status)
+    self.send(status.to_sym).count
+  end
 end
```
Comment (Lydias303) [inline app/models/company.rb:29]: I like this alot! I think it we could do some great visual stats with this!

File: app/views/admin/dashboard/index.html.erb [hunk 2]
```diff
@@ -1 +1,54 @@
...
+  <div class="confirmed-companies">
+    Confirmed Companies: <%= Company.company_count_by_status("confirmed") %>
+  </div>
+  <div class="dead-companies">
+    Companies No Longer Participating: <%= Company.company_count_by_status("dead") %>
+  </div>
+  <div class="companies-by city">
+    <h3>Top TechHire Cities</h3>
+  </div>
+</div>
```
Comment (Lydias303) [inline app/views/admin/dashboard/index.html.erb:29]: I think this is really good I like that we now have some base information for navigating what the admin can do. We will need to add some manage admin information at some point. And discuss where we would like that link to be. It might make sense to just have that as a link on the admin dashboard page as apposed to the navbar.

File: spec/factories/company.rb [hunk 3]
```diff
@@ -8,5 +8,6 @@
     email "google@email.com"
     hiring 1
     hire_count 5
+    status "uncontacted"
   end
 end
```
Comment (Lydias303) [inline spec/factories/company.rb:13]: Great to see some companies in our seed file! :sparkles: well done!

File: spec/models/company_spec.rb [hunk 4]
```diff
@@ -27,4 +27,11 @@
...
     expect(company.status).to eq("uncontacted")
   end
+
+  it "should count the number of companies by status" do
+    10.times do
+      create(:company)
+    end
+    expect(Company.company_count_by_status("uncontacted")).to eq(10)
+  end
 end
```
Comment (Lydias303) [inline spec/models/company_spec.rb:44]: Maybe at a test or multiple assertions for all the statuses.

**Comment by scottalexandra** [comment]
Added uncontacted companies to dashboard index
Add top techhire cities to dashboard
added tests
All test passing!

File: app/assets/stylesheets/application.css.scss [hunk 5]
```diff
@@ -82,3 +82,28 @@ body {
...
+  color: #147BD5;
+}
+
+.company-statistics {
+  margin-top: 75px;
+}
+
+.admin-heading h3 {
+  padding-left: 10px;
+}
```
Comment (KristaANelson) [inline app/assets/stylesheets/application.css.scss:109]: I like that the class names are good descriptions of what they cover! :sparkles:

**Comment by Lydias303** [comment]
:sparkles:

File: app/controllers/admin/dashboard_controller.rb [hunk 6]
```diff
@@ -2,6 +2,9 @@ class Admin::DashboardController < ApplicationController
   before_action :authorize!
 
   def index
-
-  end  
+    @uncontacted_companies = Company.paginate(:page => params[:page],
+                                              :per_page => 5
+                                              ).uncontacted
+                                               .order("updated_at asc")
+  end
```
Comment (KristaANelson) [inline app/controllers/admin/dashboard_controller.rb:9]: Love the paginate! Should we add the order onto all of the status scopes so that it just always orders them by updated_at? I don't think that would be a problem anywhere else if it's always ordered by updated? 

  scope :contacted, -> { where(status: "contacted").order("updated_at asc")}
  scope :uncontacted, -> { where(status: "uncontacted").order("updated_at asc")}
  scope :confirmed, -> { where(status: "confirmed").order("updated_at asc")}
  scope :dead, -> { where(status: "dead").order("updated_at asc")}

File: app/models/company.rb [hunk 7]
```diff
@@ -18,4 +18,15 @@ class Company < ActiveRecord::Base
...
+  def self.company_count_by_status(status)
+    self.send(status.to_sym).count
+  end
+
+  def self.company_count_by_city
+    self.order(:city)
+        .pluck(:city)
+        .group_by{ |city| city}
+        .inject({}) {|hash, (k,v)| hash[k] = v.size; hash}
+  end
```
Comment (KristaANelson) [inline app/models/company.rb:28]: Love the use of send and inject!! :tada: 

This would work without the selfs on line 23, and 27. Not sure what is convention (may be good to keep for readability) but I know Hound used to always yell at me for including them. 

`def self.company_count_by_status(status)`
   `send(status.to_sym).count`
 `end`

`def self.company_count_by_city`
`order(:city)`
`.pluck(:city)`
`.group_by{ |city| city}`
`.inject({}) {|hash, (k,v)| hash[k] = v.size; hash}`
`end`

File: app/views/admin/dashboard/index.html.erb [hunk 8]
```diff
@@ -1 +1,29 @@
...
+      Companies Not Yet Contacted: <%= Company.company_count_by_status("uncontacted") %>
+    </div>
+    <div class="contacted-companies">
+      Contacted Companies: <%= Company.company_count_by_status("contacted") %>
+    </div>
+    <div class="confirmed-companies">
+      Confirmed Companies: <%= Company.company_count_by_status("confirmed") %>
+    </div>
+    <div class="dead-companies">
+      Companies No Longer Participating: <%= Company.company_count_by_status("dead") %>
```
Comment (KristaANelson) [inline app/views/admin/dashboard/index.html.erb:18]: Love that you made these methods to keep the logic out of the view!!

File: app/views/application/_uncontacted_cities.html.erb [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+              </thead>
+              <tbody>
+                <% @uncontacted_companies.each do |company| %>
+                  <tr class="table-values">
+                    <td class="Organization"><%= link_to company.organization,
+                        admin_company_path(company.id) %></td>
+                    <td class="contact"><%= company.name %></td>
+                    <td class="contact-title"><%= company.title %></td>
+                    <td class="email"><%= company.email %></td>
+                    <td class="hire_count"><%= distance_of_time_in_words(company.updated_at, Time.now) %></td>
```
Comment (KristaANelson) [inline app/views/application/_uncontacted_cities.html.erb]: Looks like the hire_count class doesn't correlate to the output for that line.

Comment (scottalexandra) [inline app/views/application/_uncontacted_cities.html.erb]: oops that should be "signed_up" I will updated

File: db/seeds.rb [hunk 10]
```diff
@@ -1,125 +1,212 @@
...
+        hiring: 1,
+        hire_count: 5,
+        status: "dead"
+      )
+      i = i + 1
+    end
+  end
+end
+
+Seed.new
```
Comment (KristaANelson) [inline db/seeds.rb:224]: Love the updated seed file!

We can do this in another pull request but this could also use an example admin user and super user!

File: app/controllers/admin/dashboard_controller.rb (hunk 6 above)
Comment (KristaANelson) [inline app/controllers/admin/dashboard_controller.rb:9]: Just discussed and caught that this would be annoying in company list to have to scroll down to the bottom to see ones that were just updated. Since we are just calling this the one time, let's not worry about it and if it keeps coming up we can take a look at it then!
